### DISCUSSION TITLE
構造的なオフェンスを実現する

### DISCUSSION THREAD

**Discussion by HansRobo** [ibis-ssl/crane#392] [General]
JO2024では工数上の理由でまともなオフェンスができていなかったのでちゃんとやりたい
![image](https://github.com/ibis-ssl/crane/assets/12395284/af3f3c18-aa5e-423c-bebd-b4dd7879fdd4)

**Comment by HansRobo** [comment]
## ゲーム木の構築

何を持ってノードを分割するかはよく考えて設計しないといけない気がする。
ノード分割をすることでゲーム木として陽に考慮する要素、ゲーム木として陽に考慮せずにノード分割後にノード内で計算を行って決定する要素をよく考えないといけない。

パっと思いつく勘所は以下
- 「パスをどこで受けるか？」をゲーム木として陽に考えると状態爆発を起こすので良くない
- 一方で、次のアクションによって最適なポジショニングはかなり変わるの考慮したい。
  - 1段深く木を展開するときに前ノードの位置の修正を行う？
- あんまり深く木を展開しても状況が変わりすぎて意味なさそう
- 経由するノードで拘束されるロボット以外のことは陽に計算しなくても良い
  - むしろ不必要な拘束をかけないことであとのノードで自由に動かせるようになり、2,3手先の動きを知っている動きを作り出すことができる 
- 枝狩り
  - はじめはどうせ2,3ターン先までしか使い物にならないので下手な枝狩りはしない
  - 遷移確率を計算して枝狩り
  - 類似ノード（ほぼ同じ位置にいる2台のロボットへのパスはより受けにくい方のパスは枝狩りするなど）

**Reply by HansRobo** [reply]
「一方で、次のアクションによって最適なポジショニングはかなり変わるの考慮したい。」が結構ポイントな気がしている。
ゲーム木のノードに試合状況のコンテキストを持たせて慎重にゲーム木を展開していくよりは、コンテキストは考えず脳死でゲーム木を指定の深さまで展開したあと、展開した全ノードで天下り的にそれぞれ最適なポジショニングを考慮するのが良いかも。
これだと次のアクションがしっかりわかった状態で計算できるはず。

**Comment by HansRobo** [comment]
## ゲーム木をあまり陽に構築しない手法の検討

正直SSLはゲーム木を構築して、深く探索し、ゴールを目指すにはあまりにも流動的である。
それならば、逆に行きあたりばったりの行動を繰り返すことで試合状況をかき回し、チャンスを生み出すほうが遥かに効率的な可能性が高い。
TIGERsは意外にもこれに近い考え方のチームのようで、それほど深読みをせずにその場その場で現実的な手を次々に打っているようだ。
あまりにも素早いプレーの進め方に、その内部の処理に疑問を持ったこともあるが、この事実はかなりの説得力を持って自分の中で消化することができた。
ibisに目を向けてみると、年浅いチームであるがそのロボットの運動性能は既にベテランチームにも引けを取らない。（運動性能は高い一方で、きちんと制御できているわけではないのが玉に瑕だが）
この運動性能があれば、割と行きあたりばったりの行動で試合状況をかき乱して、生まれたチャンスをものにするだけでもかなり強くなるんじゃないかと思う今日この頃である。

**Reply by ChikuwaRobo** [reply]
👈👈👈👈👈👈👈
サッカーという否ターン性ゲームにおいては、動きの最適性よりも速さ・手数を重視すべきというのはかなり思う。
特に、動きの精密性や、細かい状況ごとのチューニングでほかチームと勝負するのは分が悪いので、まずは速さを活かした戦い方ができるように(すでにその傾向はあるけど)開発していくのがよさそう
